macros/anaUltraThreshold.py

```diff
@@ -53,143 +63,266 @@
+    for chan in range(0,vSum[vfat].GetNbinsX()):
+        for thresh in range(VT1_MAX+1,0,-1):
+            if(vSum[vfat].ProjectionY("projY",chan,chan,"").GetBinContent(thresh+1) > 1.0):
+                chanMaxVT1[0][chan]=chan
+                chanMaxVT1[1][chan]=(thresh+1)
+                break
+            pass
+        pass
+
+    #Determine Outliers (e.g. "hot" channels)
+    chanOutliers = isOutlier(chanMaxVT1[1,:])
+    for chan in range(0,len(chanOutliers)):
+        hot_channels[vfat][chan] = chanOutliers[chan]
+        pass
+
+    if options.debug:
+        print "VFAT%i Max Thresholds By Channel"%vfat
+        print chanMaxVT1
+
+        print "VFAT%i Channel Outliers"%vfat
+        chanOutliers = np.column_stack((chanMaxVT1[0,:],np.array(hot_channels[vfat]).astype(float)))
+        print chanOutliers
+        pass
+    pass
+
+# Fetch trimDAC & chMask from scurveFitTree
+import root_numpy as rp #note need root_numpy-4.7.2 (may need to run 'pip install root_numpy --upgrade')
+from qcutilities import initVFATArray
+dict_vfatTrimMaskData = {}
+if options.chConfigKnown:
+    list_bNames = ["vfatN"]
+    if not (options.channels or options.PanPin):
+        list_bNames.append("ROBstr")
+        pass
+    elif options.channels:
+        list_bNames.append("vfatCh")
+        pass
+    elif options.PanPin:
+        list_bNames.append("panPin")
+        pass
+    list_bNames.append("mask")
+    list_bNames.append("trimDAC")
+
+    try:
+        array_VFATSCurveData = rp.root2array(options.fileScurveFitTree,treename="scurveFitTree",branches=list_bNames)
+
+        #pyVersion = 1. * sys.version_info[0] + 0.1 *sys.version_info[1]
+
+        ##Initialize (key, value) pairing for dict_vfatTrimMaskData
+        #if pyVersion >= 2.7:
+        #    dict_vfatTrimMaskData = {idx:initVFATArray(array_VFATSCurveData.dtype) for idx in np.unique(array_VFATSCurveData[list_bNames[0]])}
+        #    pass
+        #else:
+        #    dict_vfatTrimMaskData = dict((idx,initVFATArray(array_VFATSCurveData.dtype)) for idx in np.unique(array_VFATSCurveData[list_bNames[0]]))
+        #    pass
+
+        #Store array_VFATSCurveData into a dict for easy access
+        #This dictionary has VFAT position as the key value, returns a structured numpy array
+        #dict_vfatTrimMaskData = {idx:initVFATArray(array_VFATSCurveData.dtype) for idx in np.unique(array_VFATSCurveData[list_bNames[0]])}
+        dict_vfatTrimMaskData = dict((idx,initVFATArray(array_VFATSCurveData.dtype)) for idx in np.unique(array_VFATSCurveData[list_bNames[0]]))
+        for dataPt in array_VFATSCurveData:
+            dict_vfatTrimMaskData[dataPt['vfatN']][dataPt[list_bNames[1]]]['mask'] =  dataPt['mask']
+            dict_vfatTrimMaskData[dataPt['vfatN']][dataPt[list_bNames[1]]]['trimDAC'] =  dataPt['trimDAC']
             pass
         pass
+    except Exception as e:
+        print '%s does not seem to exist'%options.fileScurveFitTree
```

Comment:
I don't follow, if options.fileScurveFitTree does not exist rp.root2array will generate an IOError since the option "warn_missing_tree" defaults to false:

http://scikit-hep.org/root_numpy/reference/generated/root_numpy.root2array.html#root_numpy.root2array

I don't see what other exceptions could be possible given that list_bNames is hard coded (based on the defined format of scurveFitTree).